File: 2021/6.2.php
<?php


require "6.1.php";
class DaySixPartTwo
{
    function solve($filename)
    {
        $fish = DaySixPartOne::prepareInputs($filename);
        $fish = DaySixPartOne::breedFish($fish, 256);
        echo count($fish) . "\n";
    }
}

$class = new DaySixPartTwo();
$class->solve("/2021/6test.txt");
// $class->solve("/2021/6.txt");

File: 2020/06/6.2.rb
require_relative 'helpers.rb'

module DaySixPartTwo
  module_function

  def solve(file)
    inputs = prepare_inputs(file)
    get_sum_agreed(inputs)
  end

  def get_sum_agreed(inputs)
    inputs.reduce(0) do |result, input|
      count_people = input.count("\n") + 1
      string = input.tr("\n", '')
      question_amount = string.chars.uniq.reduce(0) do |result2, letter|
        result2 += 1 if count_people == string.count(letter)
        result2
      end
      result + question_amount
    end
  end

  def prepare_inputs(file)
    Helpers.read_file(file).split("\n\n")
  end
end

# puts DaySixPartTwo.solve(__dir__ + '/6test.txt')
puts DaySixPartTwo.solve(__dir__ + '/6.txt')
